File: iosApp/PaperLess/Application/Ultility/Extensions/TagStatus.swift
//
//  TagStatus.swift
//  PaperLess
//
//  Created by Tom Salih on 27.09.25.
//

import SwiftUI
import Shared

extension TagStatus {

    static func fromFirestore(_ value: String?) -> TagStatus {
        guard let value else {
            return .available
        }
        
        switch value {
        case TagStatus.available.caseName:
            return .available
        case TagStatus.inuse.caseName:
            return .inuse
        case TagStatus.passive.caseName:
            return .passive
        default:
            return .available
        }
    }
}

extension TagStatus: @retroactive Identifiable {
    public var id: String { caseName }
    var label: String { displayName }
    var color: Color {
        switch self {
        case .available:
            return .green
        case .inuse:
            return .orange
        case .passive:
            return .error
        default:
            return .black
        }
    }
    static var all: [TagStatus] { [.available, .inuse, .passive]
    }
}

File: shared/src/commonMain/kotlin/com/tom/paperless/domain/models/enums/TagStatus.kt
package com.tom.paperless.domain.models.enums

enum class TagStatus(
    val displayName: String,
    val caseName: String,
    val caseColor: String
) {
    available(
        displayName = "Verfügbar",
        caseName = "AVAILABLE",
        caseColor = "#4CAF50" // Grün
    ),
    inUse(
        displayName = "im Einsatz",
        caseName = "IN_USE",
        caseColor = "#FF9800" // Orange
    ),
    passive(
        displayName = "Passiv",
        caseName = "PASSIVE",
        caseColor = "#9E9E9E" // Grau
    )
}